src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs

```diff
@@ -200,6 +205,10 @@ public async Task<bool> AcceptToMemoryPoolWithTime(MempoolValidationState state,
         /// <inheritdoc />
         public Task<bool> AcceptToMemoryPool(MempoolValidationState state, Transaction tx)
         {
+            // Can't properly evaluate transactions with an incomplete chain.
+            if (this.initialBlockDownloadState.IsInitialBlockDownload())
```

Comment:
Note that this will have the effect that a wallet that is still syncing will not show an incoming transaction at all until the sync is complete (obviously). I do not recall exactly what the existing behaviour is, but it might be a slight change in the user experience.

Also consider the converse - it is technically possible to submit a transaction to the network before you are fully synced as long as the selected UTXOs aren't spent in the fully synced chain. I think this would then no longer be possible as self-originated transactions don't receive special treatment as far as the local node's mempool is concerned.

If the impact of the optimisation outweighs these (admittedly probably minor) drawbacks then go for it

@fassadlr

Reply:
@zeptin, either way is fine. It's just that the sync gets a bit noisy at times due to failing mempool validation. I don't think this PR makes a huge difference optimisation-wise.